File: src/main/java/br/anhembi/funmodechild/entity/Customer.java
package br.anhembi.funmodechild.entity;

import br.anhembi.funmodechild.model.common.Address;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.time.LocalDate;
import java.util.Collection;
import java.util.List;

@Data
@Document("customers")
public class Customer implements UserDetails {

    @Id
    private String id;
    private String name;
    private String surname;
    private LocalDate birthDate;
    private String email;
    private Address address;
    private String customerIdentification;
    private String phoneNumber;
    private String password;
    private boolean enabled = true;
    private boolean accountNonExpired = true;
    private boolean credentialsNonExpired = true;
    private boolean accountNonLocked = true;

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return List.of();
    }

    @Override
    public String getUsername() {
        return this.email;
    }
}

File: src/main/kotlin/br/anhembi/funmodechild/entity/Customer.kt
package br.anhembi.funmodechild.entity

import br.anhembi.funmodechild.model.common.Address
import br.anhembi.funmodechild.model.common.LoggedUser
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.LocalDate

@Document("customers")
data class Customer(
    @Id
    val id: String? = null,
    val name: String,
    val surname: String,
    val birthDate: LocalDate,
    val email: String,
    val address: Address? = null,
    val customerIdentification: String,
    val phoneNumber: String,
    var password: String,
    val enabled: Boolean = true,
    val accountNonExpired: Boolean = true,
    val credentialsNonExpired: Boolean = true,
    val accountNonLocked: Boolean = true
) {
    fun toLoggedUser() = LoggedUser(
        id = this.id!!,
        email = this.email,
        password = this.password,
        accountNonLocked = this.accountNonLocked,
        accountNonExpired = this.accountNonExpired,
        credentialsNonExpired = this.credentialsNonExpired,
        enabled = this.enabled
    )
}
